### PR TITLE
removed trait objects from oks.rs and errors.rs

### PR DESCRIPTION
**Pull request by pchampin**
this new implementations should be more efficient,
and more faithful to the "zero-cost abstraction" mantra.

As a consequence, I had to move the 'process' method to a generic trait,
defined in util.rs and implemented by any Iterator.

A another consequence,
the 'static lifetime bound on the iterator can also be removed.

### CODE REVIEW

File: src/oks.rs [hunk 1]
```diff
@@ -1,51 +1,30 @@
 //
-// This Source Code Form is subject to the terms of the Mozilla Public
+// This Source Co²de Form is subject to the terms of the Mozilla Public
```
Comment (matthiasbeyer) [inline src/oks.rs]: This is an unneccessary change.

File: src/util.rs [hunk 2]
```diff
@@ -24,3 +24,20 @@ impl<T, E> GetOk<T> for Result<T, E> {
...
+/// Extend any Iterator with a `process` method, equivalent to a fallible for_each.
+pub trait Process<T> {
+    fn process<R: Default, E, F>(self, f: F) -> Result<R, E>
+        where F: Fn(T) -> Result<R, E>;
+}
+
+impl<I: Iterator> Process<I::Item> for I {
+    /// Process all errors with a lambda
+    fn process<R: Default, E, F>(self, f: F) -> Result<R, E>
+        where F: Fn(I::Item) -> Result<R, E> {
```
Comment (matthiasbeyer) [inline src/util.rs]: Can you please move the `{` to a new line. It improves readability IMHO.

File: src/oks.rs (hunk 1 above)
Comment (pchampin) [reply src/oks.rs]: oops, done :)

File: src/util.rs (hunk 2 above)
Comment (pchampin) [reply src/util.rs]: I agree. This is fixed too.

**Comment by pchampin** [comment]
Any plan to release the merged PRs to crates.io soon? I'd like to release my own project, and a "standard" dependency would be cleaner than my current "github" dependency... Thanks

**Comment by matthiasbeyer** [comment]
I can do this today, stay tuned!

**Comment by matthiasbeyer** [comment]
Done.

**Pull request merged**